core/slice.go

```diff
@@ -287,7 +287,19 @@ func (sl *Slice) Append(header *types.WorkObject, domTerminus common.Hash, domOr
 				sl.pEtxRetryCache.Add(block.Hash(), pEtxNew)
 				return nil, ErrSubNotSyncedToDom
 			}
-			sl.inboundEtxsCache.Add(block.Hash(), newInboundEtxs)
+			if nodeCtx == common.PRIME_CTX {
+				// write the inbound etxs to the database
+				rawdb.WriteInboundEtxs(batch, block.Hash(), newInboundEtxs)
+
+				// Copy the newInboundEtxs before using
+				newInboundEtxsCopy := make(types.Transactions, len(newInboundEtxs))
+				for i, etx := range newInboundEtxs {
+					newInboundEtxsCopy[i] = types.NewTx(etx.Inner())
```

Comment:
If you want a true copy you should do etx.Inner().copy()

Reply:
new Tx calls copy inside though